nf_core/bump_version.py

```diff
@@ -7,161 +7,208 @@
+        pipeline_obj (nf_core.utils.Pipeline): A `Pipeline` object that holds information
             about the pipeline contents and build files.
         new_version (str): The new version tag for the pipeline. Semantic versioning only.
     """
+
     # Collect the old and new version numbers
-    current_version = lint_obj.config.get("manifest.version", "").strip(" '\"")
+    current_version = pipeline_obj.nf_config.get("manifest.version", "").strip(" '\"")
     if new_version.startswith("v"):
         log.warning("Stripping leading 'v' from new version number")
         new_version = new_version[1:]
     if not current_version:
-        log.error("Could not find config variable manifest.version")
+        log.error("Could not find config variable 'manifest.version'")
         sys.exit(1)
-    log.info(
-        "Changing version number:\n  Current version number is '{}'\n  New version number will be '{}'".format(
-            current_version, new_version
-        )
-    )
-
-    # Update nextflow.config
-    nfconfig_pattern = r"version\s*=\s*[\'\"]?{}[\'\"]?".format(current_version.replace(".", r"\."))
-    nfconfig_newstr = "version = '{}'".format(new_version)
-    update_file_version("nextflow.config", lint_obj, nfconfig_pattern, nfconfig_newstr)
+    log.info("Changing version number from '{}' to '{}'".format(current_version, new_version))
 
-    # Update container tag
+    # nextflow.config - workflow manifest version
+    # nextflow.config - process container manifest version
     docker_tag = "dev"
     if new_version.replace(".", "").isdigit():
         docker_tag = new_version
     else:
         log.info("New version contains letters. Setting docker tag to 'dev'")
-    nfconfig_pattern = r"container\s*=\s*[\'\"]nfcore/{}:(?:{}|dev)[\'\"]".format(
-        lint_obj.pipeline_name.lower(), current_version.replace(".", r"\.")
-    )
-    nfconfig_newstr = "container = 'nfcore/{}:{}'".format(lint_obj.pipeline_name.lower(), docker_tag)
-    update_file_version("nextflow.config", lint_obj, nfconfig_pattern, nfconfig_newstr)
 
-    # Update GitHub Actions CI image tag (build)
-    nfconfig_pattern = r"docker build --no-cache . -t nfcore/{name}:(?:{tag}|dev)".format(
-        name=lint_obj.pipeline_name.lower(), tag=current_version.replace(".", r"\.")
-    )
-    nfconfig_newstr = "docker build --no-cache . -t nfcore/{name}:{tag}".format(
-        name=lint_obj.pipeline_name.lower(), tag=docker_tag
-    )
     update_file_version(
-        os.path.join(".github", "workflows", "ci.yml"), lint_obj, nfconfig_pattern, nfconfig_newstr, allow_multiple=True
+        "nextflow.config",
+        pipeline_obj,
+        [
+            (
+                r"version\s*=\s*[\'\"]?{}[\'\"]?".format(current_version.replace(".", r"\.")),
+                "version = '{}'".format(new_version),
+            ),
+            (
+                r"container\s*=\s*[\'\"]nfcore/{}:(?:{}|dev)[\'\"]".format(
```

Comment:
I guess you want to make this pre-DSL2 compatible? Otherwise we could just omit the whole container thing.
Same for the `environment.yml` of course.
Or maybe make it configurable, but I think it should now be able to handle DSL2 pipelines.

Reply:
Ah okay so you're just issuing a warning now if file not found ... guess that makes sense - it's not a linting command anyway :)

Reply:
Yes exactly - this code used to be super strict, only running if lint tests passed and throwing errors left right and centre if it didn't find stuff as it expected it (often leaving version numbers half-changed as a result). I made it way more lenient, so now it just does its best with each change and throws a warning if the file or pattern is not found and keeps going.

This is partly just because I don't think it needs to be so strict (we check version numbers in linting anyway) and also as you say - to future proof for DSL2. When DSL2 is well established we can just delete this bit of the code.